actionlib/test/simple_client_test.cpp

```diff
@@ -79,23 +79,17 @@ TEST(SimpleClient, easy_tests) {
 }
 
 
-void easyDoneCallback(bool * called, const SimpleClientGoalState & state,
+void easyDoneCallback(bool * called, SimpleActionClient<TestAction> * ac, const SimpleClientGoalState & state,
   const TestResultConstPtr &)
 {
-  *called = true;
+  EXPECT_TRUE(ac->getState() == SimpleClientGoalState::SUCCEEDED)
+    << "Expected [SUCCEEDED], but getState() returned [" << ac->getState().toString() << "]";
   EXPECT_TRUE(state == SimpleClientGoalState::SUCCEEDED)
     << "Expected [SUCCEEDED], but goal state is [" << state.toString() << "]";
-}
-
-void easyOldDoneCallback(bool * called, const TerminalState & terminal_state,
-  const TestResultConstPtr &)
-{
+  ros::Duration(0.1).sleep();
```

Comment:
what is this duration for?

Reply:
It's a callback, that takes some time and gives execution back to the main thread in the mean time. Without the fix, this test case `easy_callback` fails.

Reply:
See also step 2. in #155.